### PR TITLE
Get Assembly info from Main assembly for SatelliteAssembly 

### PR DESCRIPTION
**Pull request by wli3**
Instead of using the info from generating AssemblyInfo. It is not available when GenerateAssemblyInfo flag is off

**Customer scenario**

Customer build a project with AssemblyInfo.cs and has satellite assembly. The satellite assembly does not have assembly info set in the c# file, for example, no AssemblyFileVersion and no AssemblyCompany

**Bugs this fixes**

https://github.com/dotnet/sdk/issues/1276

**Workarounds, if any**

No

**Risk**

Low, the change are for satellite assemblies

**Performance impact**

Low, replacing to different msbuild target

**Root cause analysis**

lacking test coverage for edge case of satellite assembly

**How was the bug found?**

Issue filed in Github

@srivatsn could you approve this change?

### CODE REVIEW

**Comment by wli3** [comment]
@dotnet-bot test Windows_NT_FullFramework Debug

**Comment by wli3** [comment]
@dotnet-bot  test Windows_NT_FullFramework Release

File: test/Microsoft.NET.Build.Tests/GivenThatWeWanSatelliteAssembliesHaveassemblyVersion.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+using System.Diagnostics;
+using FluentAssertions;
+using System.Reflection;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWanSatelliteAssembliesHaveassemblyVersion : SdkTest
```
Comment (livarcocc) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWanSatelliteAssembliesHaveassemblyVersion.cs]: There is a typo here. It should say GivenThatWeWant, it is missing he final t. Also, HaveassemblyVersion should be HaveAssemblyVersion.

File: test/Microsoft.NET.Build.Tests/GivenThatWeWanSatelliteAssembliesHaveassemblyVersion.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        private string _mainAssembliyPath;
+        private string _satelliteAssembliyPath;
+        public GivenThatWeWanSatelliteAssembliesHaveassemblyVersion(ITestOutputHelper log) : base(log)
+        {
+            if (UsingFullFrameworkMSBuild)
```
Comment (livarcocc) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWanSatelliteAssembliesHaveassemblyVersion.cs]: This is not disabling the test in full framework, this is just skipping the constructor. Also, we have an attribute that you should put in the tests to skip them.

File: test/Microsoft.NET.Build.Tests/GivenThatWeWanSatelliteAssembliesHaveassemblyVersion.cs (hunk 1 above)
Comment (wli3) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWanSatelliteAssembliesHaveassemblyVersion.cs]: fixed -- Renamed to GivenThatWeWantSatelliteAssembliesHaveAssemblyVersion

File: test/Microsoft.NET.Build.Tests/GivenThatWeWanSatelliteAssembliesHaveassemblyVersion.cs (hunk 2 above)
Comment (wli3) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWanSatelliteAssembliesHaveassemblyVersion.cs]: fixed with [CoreMSBuildOnlyFact]

File: src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            var fileVersionInfo = FileVersionInfo.GetVersionInfo(Path.GetFullPath(RelativeFilePath));
+            CompanyName = fileVersionInfo.CompanyName;
+            LegalCopyright = fileVersionInfo.LegalCopyright;
+            Comments = fileVersionInfo.Comments;
+            FileVersion = fileVersionInfo.FileVersion;
+            ProductVersion = fileVersionInfo.ProductVersion;
+            ProductName = fileVersionInfo.ProductName;
+            FileDescription = fileVersionInfo.FileDescription;
+
+            var TryGetAssemblyVersion = FileUtilities.TryGetAssemblyVersion(Path.GetFullPath(RelativeFilePath));
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs]: variable should be called `assemblyVersion`

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets [hunk 4]
```diff
@@ -445,6 +446,52 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+      </SatelliteAssemblyAttribute>
+      <SatelliteAssemblyAttribute Include="System.Reflection.AssemblyProductAttribute" Condition="'$(_ProductNameFromTemplateFile)' != ''">
+        <_Parameter1>$(_ProductNameFromTemplateFile)</_Parameter1>
+      </SatelliteAssemblyAttribute>
+      <SatelliteAssemblyAttribute Include="System.Reflection.AssemblyTitleAttribute" Condition="'$(_FileDescriptionFromTemplateFile)' != ''">
+        <_Parameter1>$(_FileDescriptionFromTemplateFile)</_Parameter1>
+      </SatelliteAssemblyAttribute>
+      <SatelliteAssemblyAttribute Include="System.Reflection.AssemblyVersionAttribute" Condition="'$(_AssemblyVersionFromTemplateFile)' != ''">
+        <_Parameter1>$(_AssemblyVersionFromTemplateFile)</_Parameter1>
+      </SatelliteAssemblyAttribute>
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets]: I would consider having the task return these items and renaming it to GetAssemblyAttributes. It will be more efficient with less code in XML to maintain.

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets [hunk 5]
```diff
@@ -445,6 +446,52 @@ Copyright (c) .NET Foundation. All rights reserved.
     </Csc>
   </Target>
 
+  <Target Name="_GetAssemblyInfoFromTemplateFile">
+
+    <PropertyGroup>
+      <_CoreGenerateSatelliteAssembliesTemplateFile>$(IntermediateOutputPath)$(TargetName)$(TargetExt)</_CoreGenerateSatelliteAssembliesTemplateFile>
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets]: Consider removing this variable and just giving the task parameter a more descriptive name. `RelativePathToTemplateFile`

**Comment by nguerrera** [comment]
Approach looks good. My comments are minor. Nice work.

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets (hunk 5 above)
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets]: It's actually not necessarily relative, just possibly relative. IntermediateOutputPath is often setup to be a full path.

File: test/Microsoft.NET.Build.Tests/GivenThatWeWantSatelliteAssembliesHaveassemblyVersion.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            satelliteAssembliyFileVersioninfo.FileDescription.Should().Be(mainAssembliyFileVersioninfo.FileDescription);
+        }
+
+        [CoreMSBuildOnlyFact]
+        public void It_should_produce_same_SatelliteAssemblie_AssemblyVersions_as_main()
+        {
+            RestoreAndBuildTestAssets();
+
+            var mainAssembliy = AssemblyName.GetAssemblyName(_mainAssembliyPath);
+            var satelliteAssembliy = AssemblyName.GetAssemblyName(_satelliteAssembliyPath);
```
Comment (nguerrera) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantSatelliteAssembliesHaveassemblyVersion.cs]: typo: mainAssembly, satelliteAssembly (in variable and field names)

File: src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs (hunk 3 above)
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs]: Fixed, changed to assemblyVersion

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets (hunk 4 above)
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets]: Fixed, moved all these logic to csharp code in task

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets (hunk 5 above)
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets]: _CoreGenerateSatelliteAssembliesTemplateFile is removed, the parameter is called PathToTemplateFile

File: test/Microsoft.NET.Build.Tests/GivenThatWeWantSatelliteAssembliesHaveassemblyVersion.cs (hunk 6 above)
Comment (wli3) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantSatelliteAssembliesHaveassemblyVersion.cs]: all fixed

**Review by nguerrera — APPROVED** [review]
